src/NUnitFramework/tests/Internal/DefaultSuiteBuilderTests.cs

```diff
@@ -0,0 +1,37 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using NUnit.TestData;
+
+namespace NUnit.Framework.Internal
+{
+    [TestFixture]
+    public static class DefaultSuiteBuilderTests
+    {
+        [Test]
+        public static void ClassesCanBeImpliedToBeFixtures()
+        {
+            Assert.That(new Internal.Builders.DefaultSuiteBuilder().CanBuildFrom(typeof(ImpliedFixture)));
```

Comment:
This tells us that it *says* it can build a fixture, but can we does what it actually *does*? I.e, that the test suite it returns contains a fixture for the type `DefaultSuiteBuilderTests`?

Reply:
I usually try to write the smallest possible micro-test for the thing I'm testing. IImplyFixture is only used within the `CanBuildFrom` method. The `BuildFrom` method does the same thing no matter how the fixture is identified.

Adding a call to `BuildFrom` would reassure you but would also be a test of something else that was not impacted by this issue. 

I think your comment reflects the part that we don't have any other tests of DefaultSuiteBuilder, and you'd like to see more. I could do that as a part of this PR even though it goes against my "smallest possible diff" initiative. 😄 

Your comment and my approach __may__ also reflect a different view of the nature of these tests: black box versus white box. Since I __know__ that the implementation only makes use of the interface in the `CanBuild` method, I can write a micro-test for that alone. I assume (blindly and wrongly) that some functional test will catch less localized errors. IOW, so long as we don't have separate functional and integration tests, there is always a temptation to write unit tests that test a few things at once.

That said, I'm glad to write a few more tests if you like!

Reply:
Sure, splitting it into smaller microtests seems even better to me! I would like that, thank you!